### ISSUE TITLE
Finding a dashboard to our github CI

### ISSUE THREAD

**Issue by wainersm** [kata-containers/kata-containers#9892]
**Which feature do you think can be improved?**

Once upon a time, the Jenkins' dashboard was used to easily identify unstable CI jobs, discuss on CoCo's meeting and hopefully take actions. With the CI migration to github workflow we lost that dashboard and Github doesn't provide one.

I've been using a python script (see https://github.com/kata-containers/kata-containers/issues/9506) to collect statistics out of the nightly jobs to guide me on chopping  unstable ones. While the script works fine for that purpose, it doesn't help on the CoCo's meetings for sharing the CI status nor it provides more detailed information (e.g. failure trend, view by PR...etc).

**How can it be improved?**

Find a off shelf dashboard or implement one ourselves.

Some options that I've discussed with @ariel-adam , @portersrc and others on CoCo's CI meetings:

 * Buy (or get for free!) a subscription to use [DataDog's dashboard for Github actions](https://www.datadoghq.com/blog/datadog-github-actions-ci-visibility)
 * Implement our own in-house dashboard
 * Reach out Red Hat's QA team to assess their tooling and what could be used (if any) 

Please, if you have other options let's us know.

Who's involved:
 * @wainersm 
 * @portersrc 
 
Breaking down in work items:

- [x] Call for participation
- [ ] Assess each option
- [ ] Take a decision
- [ ] Implement and roll out

**Comment by wainersm** [comment]
Call for participation posted in https://katacontainers.slack.com/archives/C879ACQ00/p1719004516278219

**Comment by portersrc** [comment]
Bumping the link to the [Datadog application](https://www.datadoghq.com/partner/open-source/?_gl=1*vnvwpo*_gcl_au*NjcwMDUzMDgzLjE3MTc2MTIyNjI.*_ga*MTUwNzgzMzU3Ni4xNzE3NjEyMjYy*_ga_KN80RDFSQK*MTcxODgxNTU0NC40LjAuMTcxODgxNTU0NC4wLjAuNzAxOTQ4ODE1*_fplc*MmpPNjFUd3g5TmMwTEZ3WGhsOE1vZFJvRnVaTXZQSWElMkZaSiUyRlZoUE1JNEpGN25KTDIwb2JndE9YUjRrJTJGQWg2YiUyRnVMTXpPSlpDd0w0RUZlbHlveW5jOHByRG54dHFwc2pFZ0NIeXJ0cHlqZGclMkZyNTlKYUdGeUxpNURIYms1ZyUzRCUzRA) for free support for open source projects.

Links to tools I checked when I did a bake-off months back for coco:
https://meercode.io/
https://github.com/otto-de/gitactionboard
https://github.com/pystol/badgeboard
https://gitboard.io/
https://www.datadoghq.com/product/ci-cd-monitoring/

As some recall, I free-trial'd the datadog option and demo'd for us, and I think it could work. The others in that list didn't look like good options for us. But both of the other paths @wainersm mentioned could work well (in-house or leveraging a redhat tool).

**Comment by fidencio** [comment]
> As some recall, I free-trial'd the datadog option and demo'd for us, and I think it could work.

@portersrc, please, correct me if I'm mistaken, but could DataDog provide a public dashboard?  IIRC, having to have an account / password that only a few people could monitor was one of the reasons we ended up not pursuing this.

I'd be fine on having this, and then getting someone (@ariel-adam ?) to post daily updates about the weather, but it'd need to be someone / some team responsibility to do so, which is not ideal ... but is acceptable.

**Comment by fidencio** [comment]
https://octolense.com/ -- this seems promising

**Comment by portersrc** [comment]
> correct me if I'm mistaken, but could DataDog provide a public dashboard?
That's right. I think we pulled the brakes on datadog because of this. My view is, if we apply to for datadog, we should include this concern in our comments. (For example, could we have an admin login for maintainers, and a general, shareable read-only login for anyone in the open source community. Surely they've dealt with this before.)

octolense is another good example. One doubt I had with it is it's free because it's in public beta. I wasn't sure what to expect after that ends.

**Comment by fidencio** [comment]
> octolense is another good example. One doubt I had with it is it's free because it's in public beta. I wasn't sure what to expect after that ends.

I agree, and I think it's worth directly checking with them if they'd be willing to provide any sponsorship for OpenSource projects.

**Comment by fidencio** [comment]
Another option crossing my mind here, but then for something more custom, would be https://bitergia.com/
We already have a community dashboard coming from them, in partnership with OpenInfra Foundation: https://katacontainers.biterg.io/app/dashboards#/view/Overview

Maybe we could rely on @ildikov to check with them whether they could help us with the CI dashboard.

**Comment by ildikov** [comment]
@fidencio Thank you for bringing up your idea. As far as I know the Bitergia dashboard and analytics don't support the CI dashboard use case. Their platform and services focus on team work and collaboration metrics, as opposed to SW health.

I will double check this, and report back on this issue if I find anything related in the Bitergia offerings.

**Comment by wainersm** [comment]
Hi @fidencio !

Thanks for the inputs!

> > octolense is another good example. One doubt I had with it is it's free because it's in public beta. I wasn't sure what to expect after that ends.
> 
> I agree, and I think it's worth directly checking with them if they'd be willing to provide any sponsorship for OpenSource projects.

Can @portersrc and I approach octolense (and other providers) and talk on behalf of Kata Containers?

**Comment by wainersm** [comment]
Updates:
 * @portersrc created a requirements document available in https://docs.google.com/document/d/11-hF13zII48rspxep4I1TCv_TO8nlZtuJM0vMUca1Gw/edit#heading=h.v7jnqo9muyh4
 * @ildikov at al. to have a meeting with Datadog to discuss their dashboard and the possibility to use in our project
 * I played with https://grafana.com 's   [github-datasource](https://grafana.com/grafana/plugins/grafana-github-datasource). I couldn't find a way to generate jobs charts; it seems that it will ingest data only up to workflows (i.e. not able to break workflows down to jobs level).

**Comment by ariel-adam** [comment]
@wainersm how are we doing on this one :-) ?

**Comment by portersrc** [comment]
Hey @ariel-adam, since @wainersm's last comment above, we shifted gears to do something ourselves that's basic and covers some needs, rather than use a pre-existing tool. Our CI group has made progress. Right now it captures the results from @wainersm's script [here](https://github.com/kata-containers/kata-containers/issues/9506#issuecomment-2168522061) and displays them. The two main things we need to answer in the [CI meeting](https://docs.google.com/document/d/1gVuJXZzdyZcBg0Vje6n3-uL-SYdgxHzzEw9moU2ycoE/edit) tomorrow (Wednesday) are: how do we want to host/deploy it (and note that it requires a github API token for making requests); and if we're OK with the current approach, are there other any other major things we want in this first version?

**Comment by ariel-adam** [comment]
Sounds like solid progress @portersrc 
Appreciate the detailed update

**Comment by portersrc** [comment]
The coco CI folks have presented now to the general community and gotten feedback. It seems this [demo](https://portersrc.github.io/) is sufficient for us to go ahead. We seemed to have tacit agreement that we could use the kata-containers's github pages to host it. Should I create a separate issue for someone to create a kata-containers.github.io repo, or who is the right person or people to @ here to help make this happen?

**Comment by stevenhorsman** [comment]
Resolved
